lib/matplotlib/legend_handler.py

```diff
@@ -427,6 +427,32 @@ def create_artists(self, legend, orig_handle,
         return [legline]
 
 
+class HandlerPatchCollection(HandlerPatch):
+    """
+    Handler for `.PatchCollection` instances.
+    """
+    def _default_update_prop(self, legend_handle, orig_handle):
+        lw = orig_handle.get_linewidths()[0]
+        dashes = orig_handle._us_linestyles[0]
+        facecolor = orig_handle.get_facecolor()[0]
+        edgecolor = orig_handle.get_edgecolor()[0]
```

Comment:
Should probably do the same as `HandlerPolyCollection` and handle the empty case (i.e. not require the index 0 element to exist)